### PR TITLE
UP-1689 get last device

### PR DESCRIPTION
**Pull request by benoit-grg**
- removed customer id query as it's no longer necessary for the front end to have it
- changed the logic behind querying for the latest hash according to UP-1689
- added API endpoint for this functionality

### CODE REVIEW

File: src/main/scala/com/ubirch/webui/services/ApiDevices.scala [hunk 1]
```diff
@@ -552,13 +552,40 @@ class ApiDevices(implicit val swagger: Swagger)
...
     }
   }
 
+  val getLastHash: SwaggerSupportSyntax.OperationBuilder =
+    (apiOperation[LastHash]("getLastHashDevice")
+      summary "Get the last hash produced by a device"
+      description "Get the last hash produced by a device"
+      tags "Devices"
+      parameters (
+      swaggerTokenAsHeader,
```
Comment (lalottacontinua) [inline src/main/scala/com/ubirch/webui/services/ApiDevices.scala:569]: doesn't need a description?

File: src/main/scala/com/ubirch/webui/services/ApiDevices.scala [hunk 2]
```diff
@@ -552,13 +552,40 @@ class ApiDevices(implicit val swagger: Swagger)
...
+  get("/lastHash/:id") {
+    logger.debug(s"devices: get(/lastHash/$getHwDeviceId)")
+    whenLoggedInAsUser { (userInfo, user) =>
+      val hwDeviceId = getHwDeviceId
+      implicit val realmName: String = userInfo.realmName
+
+      DeviceFactory.getByHwDeviceId(hwDeviceId) match {
+        case Left(_) => stopBadUUID(hwDeviceId)
+        case Right(device) =>
+          if (device.isUserAuthorizedBoolean(user)) {
```
Comment (lalottacontinua) [inline src/main/scala/com/ubirch/webui/services/ApiDevices.scala]: the method could become named isUserAuthorized (without Boolean)

**Comment by lalottacontinua** [comment]
cannot say anything about the Gremlin stuff, but looks as it has become more clear and efficient. ;)

File: src/main/scala/com/ubirch/webui/services/ApiDevices.scala (hunk 2 above)
Comment (benoit-grg) [reply src/main/scala/com/ubirch/webui/services/ApiDevices.scala]: Good idea, changed :)

File: src/main/scala/com/ubirch/webui/services/ApiDevices.scala (hunk 1 above)
Comment (benoit-grg) [reply src/main/scala/com/ubirch/webui/services/ApiDevices.scala:569]: True, it's quite minimalistic, will change that

**Pull request merged**